src/main/java/com/uber/cadence/samples/hello/HelloAsyncActivityCompletion.java

```diff
@@ -30,95 +31,97 @@
 import java.util.concurrent.ForkJoinPool;
 
 /**
- * Demonstrates an asynchronous activity implementation.
- * Requires a local instance of Cadence server running.
+ * Demonstrates an asynchronous activity implementation. Requires a local instance of Cadence server
+ * running.
  */
 public class HelloAsyncActivityCompletion {
 
-    private static final String TASK_LIST = "HelloAsyncActivityCompletion";
+  static final String TASK_LIST = "HelloAsyncActivityCompletion";
 
-    public interface GreetingWorkflow {
-        /**
-         * @return greeting string
-         */
-        @WorkflowMethod(executionStartToCloseTimeoutSeconds = 15, taskList = TASK_LIST)
-        String getGreeting(String name);
-    }
+  public interface GreetingWorkflow {
+    /** @return greeting string */
+    @WorkflowMethod(executionStartToCloseTimeoutSeconds = 15, taskList = TASK_LIST)
+    String getGreeting(String name);
+  }
+
+  /** Activity interface is just a POJI */
+  public interface GreetingActivities {
+    @ActivityMethod(scheduleToCloseTimeoutSeconds = 10)
+    String composeGreeting(String greeting, String name);
+  }
+
+  /** GreetingWorkflow implementation that calls GreetingsActivities#printIt. */
+  public static class GreetingWorkflowImpl implements GreetingWorkflow {
 
     /**
-     * Activity interface is just a POJI
+     * Activity stub implements activity interface and proxies calls to it to Cadence activity
+     * invocations. As activities are reentrant only a single stub can be used for multiple activity
+     * invocations.
      */
-    public interface GreetingActivities {
-        @ActivityMethod(scheduleToCloseTimeoutSeconds = 10)
-        String composeGreeting(String greeting, String name);
+    private final GreetingActivities activities =
+        Workflow.newActivityStub(GreetingActivities.class);
+
+    @Override
+    public String getGreeting(String name) {
+      // This is blocking call that returns only after activity is completed.
```

Comment:
is blocking -> is a blocking
after activity is completed -> after the activity is complete OR after the activity has completed.